std/range/package.d

```diff
@@ -8606,6 +8606,385 @@ if (isForwardRange!Source && hasLength!Source)
+starting from the first element of the source and ending the first chunk at the first index in
+endIndexes, not including this last element. Both ranges must be $(REF_ALTTEXT input ranges, isInputRange, std,range,primitives).
+The indexes must be in ascending order.
+
+If the `source` range is shorter than the the provided indexes, the specified Indexes
+are shortened to match the length of the source range.
+
+If the `source` range is longer than the last index provided, all extra elements will
+form one new chunk.
+
+If `Source` and `Indexes` are forward ranges, the resulting range will be a forward range as well.
+Otherwise, the resulting chunks will be input ranges using the same input. The chunk will shrink
+with every `front` call and `popFront` will invalidate the previous `front`.
+
+Params:
+    source = range from which the chunks will be selected
+    endIndexes = range specifying the indexes of the elements that will not be included in the current chunk
+
+Returns: Range of chunks of variable lengths
+*/
+auto chunks(Source, Indexes)(Source source, Indexes endIndexes)
+if (isInputRange!Source && isInputRange!Indexes && is(ElementType!Indexes : size_t))
+{
+    struct Chunks(Source, Indexes)
+    {
+        static if (isForwardRange!Source && isForwardRange!Indexes)
+        {
+            this(Source source, Indexes endIndexes, size_t offset = 0, UseMaxLength maxLen=UseMaxLength(false))
+            {
+                import std.algorithm.sorting : isStrictlyMonotonic;
+                assert(endIndexes.isStrictlyMonotonic, "The indexes must be in a strictly increasing order");
+
+                _source = source;
+                _endIndexes = endIndexes;
+                _offset = offset;
+                _maxLen = maxLen;
+                if (!_source.empty)
+                    _isEmpty = false;
+                else
+                    _isEmpty = true;
+            }
+
+            /// Input range primitives. Always present.
+            @property bool empty()
+            {
+                return _isEmpty;
+            }
+
+            /// Ditto
+            @property auto front()
+            {
+                assert(!empty, "Attempting to fetch the front of an empty range");
+
+                if (_endIndexes.empty && !_source.empty)
+                    return _source.save;
+
+                assert(_endIndexes.front != 0, "Attempting to create a chunk of length 0");
+
+                return _source.save.take(_endIndexes.front()-_offset);
+            }
+
+            /// Ditto
+            void popFront()
+            {
+                assert(!empty, "Attempting to popFront an empty range");
+
+                // source range is longer than index
+                if (_endIndexes.empty && !_source.empty)
+                {
+                    _isEmpty = true;
+                    return;
+                }
+
+                _source.popFrontN(_endIndexes.front()-_offset);
```

Comment:
protect this line with `assert(_endIndexes.front >= _offset)`. Also, remove the parentheses for front

Reply:
Moot.